app/code/AstoundDRudenko/Install/Setup/Patch/Data/RelocateTestAttributeData.php

```diff
@@ -0,0 +1,216 @@
+    private $testDataAttribute;
+
+    /**
+     * RelocateTestAttributeData constructor.
+     * @param InstallReindexer $installReindexer
+     * @param ProductResource $productResource
+     * @param EavConfigFactory $eavConfigFactory
+     * @param ModuleDataSetupInterface $moduleDataSetup
+     * @param EavSetupFactory $eavSetupFactory
+     */
+    public function __construct(
+        InstallReindexer $installReindexer,
+        ProductResource $productResource,
+        EavConfigFactory $eavConfigFactory,
+        ModuleDataSetupInterface $moduleDataSetup,
+        EavSetupFactory $eavSetupFactory
+    ) {
+        $this->moduleDataSetup = $moduleDataSetup;
+        $this->eavSetupFactory = $eavSetupFactory;
+        $this->eavConfigFactory = $eavConfigFactory;
+        $this->productResource = $productResource;
+        $this->installReindexer = $installReindexer;
+    }
+
+    /**
+     * Do Upgrade
+     *
+     * @return void
+     */
+    public function apply()
+    {
+        $eavSetup = $this->eavSetupFactory->create(['setup' => $this->moduleDataSetup]);
+
+        $this->connection = $this->moduleDataSetup->getConnection();
+        $oldTable = $this->getTestDataAttribute()->getBackendTable();
+
+        $eavSetup->updateAttribute(
+            Product::ENTITY,
+            Config::TEST_ATTRIBUTE_CODE,
+            'backend_type',
+            'text'
+        );
+
+        $fields = [
+            AttributeInterface::ATTRIBUTE_ID,
+            Product::STORE_ID,
+            'value',
+            $this->productResource->getLinkField()
+        ];
+
+        $this->testDataAttribute = $this->getTestDataAttribute();
+        $newTable = $this->testDataAttribute->getBackendTable();
+
+        $select = $this->getAttributeDataSelect($oldTable, $fields);
+
+        $this->insertAttributeData($select, $newTable, $fields);
+        $this->deleteAttributeData($select, $oldTable);
+
+        $productIds = $this->getUpdatedProductIds($newTable);
+
+        $this->installReindexer->reindexProductAttributes($productIds);
+    }
+
+    /**
+     * Query to update attribute data
+     * @param string $newTable
+     * @return array
+     */
+    private function getUpdatedProductIds(string $newTable) :array
+    {
+        $productIdsSelect = $this->connection->select()
+            ->from(
+                $this->moduleDataSetup->getTable($newTable),
+                [$this->productResource->getLinkField()]
+            )->where(
+                AttributeInterface::ATTRIBUTE_ID . ' = ?', $this->testDataAttribute->getId()
+            );
+
+        return $this->connection->fetchCol($productIdsSelect);
+    }
+
+    /**
+     * Query to delete attribute data
+     * @param Select $select
+     * @param string $oldTable
+     */
+    private function deleteAttributeData(Select $select, string $oldTable)
+    {
+        $deleteQuery = $this->connection->deleteFromSelect($select, $oldTable);
+        $this->connection->query($deleteQuery);
```

Comment:
Every method should return something, if nothing to be returned - return $this.
Exceptions are some void method like init(), prepare(), etc.